src/main/ruby/core/truffle/interop.rb

```diff
@@ -25,16 +25,25 @@ def self.export_method(name)
       export(name.to_s, Object.method(name.to_sym))
     end
 
-    def self.object_keys(object)
+    def self.object_keys(object, internal)
       if object.is_a?(Hash)
-        object.keys.map(&:to_s)
+        keys = object.keys.map(&:to_s)
       else
-        object.instance_variables.map { |ivar|
+        keys = object.instance_variables.map { |ivar|
           ivar = ivar.to_s
           ivar = ivar[1..-1] if ivar.start_with?('@')
           ivar
         }
       end
+      if internal
+        object.instance_variables.each do |ivar|
+          ivar = ivar.to_s
+          if ivar.start_with?('@')
```

Comment:
Isn't it a guarantee that `instance_variables` only returns names with `@`?

Reply:
I don't know - but there was existing code above this checking for it.

Reply:
If you remove the checks it still works. Not sure what to do.

Reply:
I would remove it, it's a bug if `instance_variables` returns internal fields.